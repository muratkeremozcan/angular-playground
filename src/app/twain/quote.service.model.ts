import { Observable } from 'rxjs';

export interface QuoteServiceModel {
  getQuote: () => Observable<string>;
}
