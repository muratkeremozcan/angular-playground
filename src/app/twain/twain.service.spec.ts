import { SpectatorHttp, createHttpFactory, HttpMethod } from '@ngneat/spectator/jest';
import { TwainService } from './twain.service';
import { Quote } from './quote';

describe('TwainService', () => {
  let spectator: SpectatorHttp<TwainService>;
  let twainService: TwainService;
  let assertion;
  let expectedQuote: Quote;

  const createHttp = createHttpFactory(TwainService);

  beforeEach(() => {
    spectator = createHttp();
    twainService = spectator.service;

    expectedQuote = {
      id: 1,
      quote: 'abc'
    };
  });

  it('should getQuote', () => {
    twainService.getQuote().subscribe(
      (quote) => (assertion = quote),
      () => { throw new Error() }
    );

    spectator.expectOne('api/quotes/1', HttpMethod.GET).flush(expectedQuote);

    expect(assertion).toEqual(expectedQuote.quote);

    expect(expectedQuote).toMatchSnapshot();
  });
});
