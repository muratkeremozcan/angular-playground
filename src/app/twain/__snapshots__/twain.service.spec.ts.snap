// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwainService should getQuote 1`] = `
{
  "id": 1,
  "quote": "abc",
}
`;
